pkg/smart_wallet/init_code.go

```diff
@@ -1,92 +1,47 @@
 package smart_wallet
 
 import (
-	"context"
-	"errors"
 	"fmt"
 	"log/slog"
 
-	"github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
-	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/shopspring/decimal"
 )
 
-var (
-	res, ok                  = entryPointABI.Errors["SenderAddressResult"]
-	senderAddressResultError = boolMust(res, ok)
-)
-
-func IsAccountDeployed(ctx context.Context, provider ethereum.ChainStateReader, swAddress common.Address) (bool, error) {
-	byteCode, err := provider.CodeAt(ctx, swAddress, nil)
-	if err != nil {
-		return false, fmt.Errorf("failed to check if smart account is already deployed: %w", err)
-	}
-
-	// assume that the smart account is deployed if it has non-zero byte code
-	return len(byteCode) != 0, nil
-}
+func GetInitCode(smartWalletConfig Config, ownerAddress common.Address, index decimal.Decimal) ([]byte, error) {
+	var initCode []byte
+	var err error
 
-func GetAccountAddress(ctx context.Context, provider ethereum.ContractCaller, config Config, entryPointAddress, owner common.Address, index decimal.Decimal) (common.Address, error) {
-	initCode, err := GetInitCode(config, owner, index)
-	if err != nil {
-		return common.Address{}, fmt.Errorf("failed to build initCode middleware: %w", err)
+	switch typ := *smartWalletConfig.Type; typ {
```

Comment:
Now that the function is moved from `userop` package and is no longer private, we have no nil pointer "safety guards" and therefore it should have an associated default value. So, for example, if `smartWalletConfig.Type` is `nil`, the function interprets it as KernelType.

Reply:
I don't think the package should specify the default value for a smart wallet type. Instead, we can check the Type not to be nil.

Reply:
03b2806414bea4f017c757d5a577c632ea17f7b9